grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy

```diff
@@ -161,10 +163,12 @@ class ReportController {
     def exportBinLocation() {
         long startTime = System.currentTimeMillis()
         log.info "Export by bin location " + params
+        Boolean hasRoleFinance = userService.hasRoleFinance(AuthService.getCurrentUser())
         Location location = Location.get(session.warehouse.id)
-        List binLocations = inventoryService.getQuantityByBinLocation(location)
-        def products = binLocations.collect { it.product.productCode }.unique()
-        binLocations = binLocations.collect {
+        List<BinLocationItem> binLocations = inventoryService.getQuantityByBinLocation(location)
+        List<String> products = binLocations.collect { it.product.productCode }.unique()
```

Comment:
I was so confused by these two lines. I thought the unique products were then going to be used to manipulate the binLocations somehow and was going to suggest creating another service method to return the result.

Then I realized products is only used to return a productCount. So I would recommend just returning the count here and setting a default value.
```
int productCount = binLocations.collect { it.product.productCode }.unique()?.size() ?: 0
```
There's gotta be a more Groovier way of doing this ^^ no? I spend a few minutes and couldn't find what I was looking for, but it feels like something we should be able to do more succinctly.

Reply:
I agree with you, initially I didn't pay much attention to that, since I did only a cosmetic change (def -> type) for that line.
I think I can try something with `.groupBy`.

Reply:
I went with the `.groupBy` approach. I hope it's cleaner now.